hermes-api/src/main/java/pl/allegro/tech/hermes/api/constraints/Names.java

```diff
@@ -1,5 +1,16 @@
 package pl.allegro.tech.hermes.api.constraints;
 
 public final class Names {
-    public static final String ALLOWED_NAME_REGEX = "[a-zA-Z0-9_.-]+";
+    private static final String ALLOWED_CHARACTERS = "[a-zA-Z0-9_]";
+    private static final String DELIMITER = "\\.";
+
+    public static final String ALLOWED_NAME_REGEX = ALLOWED_CHARACTERS + "+";
+
+    public static final String ALLOWED_GROUP_NAME_REGEX =
+            ALLOWED_NAME_REGEX + "(" + DELIMITER + ALLOWED_NAME_REGEX + ")*";
+
+    public static final String ALLOWED_TOPIC_NAME_REGEX = ALLOWED_NAME_REGEX;
+
+    public static final String ALLOWED_SUBSCRIPTION_NAME_REGEX = ALLOWED_NAME_REGEX;
```

Comment:
Subscription name also can contain implicit namespace separated by dots. Although, we do not force it to use it by Hermes users but in Allegro we are using it widely as it is more verbose name. For example we have hundreds of subscriptions named in convention `pl.allegro.{domain}.{context}`, for instance `pl.allegro.advertising.inventory`. Therefore, please do not restrict dots usage in subscription name here and and in Hermes-console.

Reply:
Hi, thanks for review. I've addressed your comment in latest commit (regexp for subscription name validation).